### ISSUE TITLE
Is lmdb.dll compiled in Debug mode?

### ISSUE THREAD

**Issue by marcotod1410** [CoreyKaylor/Lightning.NET#171]
In some Windows production environments, we observed that a specific Put operation makes the entire application crash in an uncontrollable way using the latest version of LightningDB and .NET 8.
This is the code for reference: 
```
using var env = new LightningEnvironment("path\\to\\data")
{
    MapSize = 100 * 1024 * 1024,
    MaxDatabases = 1
};
env.Open();

var byteKey = Encoding.UTF8.GetBytes("shard-StoragePubSub-all");
var byteValue = new byte[] { 10, 26, 10, 19, 10, 1, 55, 10, 2, 55, 51, 10, 2, 55, 53, 10, 2, 57, 50, 18, 2, 8, 2, 16, 11, 34, 1, 66 };

using (var tx = env.BeginTransaction())
using (var db = tx.OpenDatabase("ddata"))
{
    var putResult = tx.Put(db, byteKey, byteValue);  // crash!
    var result = tx.Commit();
}
```
Keep in mind that we were able to reproduce the crash only with a already created .mdb file taken from production. Let me know if you need it for testing.

While debugging, we discovered that the application crashes (exiting with code 3) on a failed assertion from lmdb.dll since the following message is printed on the console:
`mdb.c:2741: Assertion 'mp->mp_pgno != pgno' failed in mdb_page_touch()`
which is not expected in my opinion, since the lmdb.dll library should be built in Release mode and assertions should not be enabled.

I've managed to build the lmdb library myself in version 0.9.29 (using [this script](https://github.com/lmdbjava/lmdbjava/blob/master/cross-compile.sh)). With the rebuilt dll in Release mode, the application does not crash anymore and the Put operation returns a success status code. Conversely, I've built the library in Debug mode, by adding `CFLAGS="-DDEBUG"`, and the assertion is effectively executed, leading to the application crash.
Of course I'm not questioning the correctness of the operation since that assertion must mean something and for this reason I'll conduct an in-depth study of possible consequences over the fresh written database.

That being said, I've got a couple of questions:
- Are native libraries bundled in Lightning.NET built in Debug mode purposefully?
- If not so, would it make sense to also update the library versions, since the latest version is 0.9.33 and the bundled version is 3 years overdue?

**Comment by CoreyKaylor** [comment]
The native compile script is located here. https://github.com/CoreyKaylor/Lightning.NET/blob/master/lmdb/compile-lmdb-macos.sh

In the last few releases the script was added as the number of targets grew fairly substantially and the toolchain dependencies were a bit crazy, so it looks like you're right that it is built in debug mode. Are you up for a pull request?

**Comment by CoreyKaylor** [comment]
Just an initial glance. It looks like the changes necessary would be to update the repo in the script to instead point at the primary openldap repository as it seems the mirror on github is falling out of sync with the core repo. Separately, I'm not sure what the status is on the "growing disk size for windows" commits. That's been a sensitive topic for folks using this lib, but also doesn't seem like it is supported in several years unless I missed something, so maybe that's worth a separate discussion.

**Comment by marcotod1410** [comment]
I'm available for opening a PR. Are you talking about replacing only the binaries for Windows environments (including the autoresize dll) or for each of the environments listed? Unfortunately I'm not able to build the library for Apple environments, since I'm building the libraries in a WSL environment and it looks like that XCode is required to build for these specific envs. I'm able to build for Linux and Android though.
Also, I don't know if this issue is present in other environments, but I'll give it a try at least in a Linux environment with the same test and see what happens.

I see your concerns in upgrading the binaries and I currently don't know the status of this feature from the LMDB side, so I'll stick with the 0.9.29 version for now.

**Comment by CoreyKaylor** [comment]
I think if you just update the script for the PR, that would be good enough. I can deal with the toolchain dependencies and producing the binaries.

**Comment by CoreyKaylor** [comment]
Fixed in 84385ac
